config/routes.rb

```diff
@@ -1,8 +1,14 @@
 Rails.application.routes.draw do
   
-  root to: 'users#index'
+  root "users#index"
 
-  resources :users, only: [:index, :show] do
+  get '/posts/new', to: 'posts#new', as: 'create_post'
+  post '/posts/new', to: 'posts#create'
+  get '/users/:user_id/posts/:id', to: 'posts#show', as: 'post'
+  post '/users/:user_id/posts/:id/', to: 'posts#create_comment'
+  post '/users/:user_id/posts/:id/likes', to: 'posts#create_like', as: 'like'
+
+  resources :users, only: [:index, :show] do 
     resources :posts, only: [:index, :show]
   end
```

Comment:
- [ ] In fulfillment of the suggested changes above, the comments and likes routes on lines 8 and 9 should be removed or refactored to point to the likes and comments controllers instead of the posts controller.

- **[OPTIONAL]** We can also wrap the comments and likes route within the posts.

  ```
  ... 
    resources :posts, only: [:index, :show] do
      resources :comments, only: [:create] 
      resources :likes, only: [:create]
    end
  ...
  ```